### PR TITLE
feat(doc): adjust Markdown pipe-table column widths from cell content

### PR DESCRIPTION
**Pull request by ejjcc**
## Summary

`docs +create` and `docs +update --mode overwrite` currently upload Markdown tables that render with equal column widths (e.g. 244/244/244 for a 3-column table) regardless of how uneven the cell content is. Users have to manually drag columns in the UI to get a readable layout.

This PR adds a local pipe-table parser and a content-aware width-ratio calculator. After the MCP create/update call succeeds, the CLI walks the document's table blocks in order, matches each against the corresponding Markdown table by column count, and PATCHes \`update_grid_column_width_ratio\` with ratios summing to 100.

## Changes

- New: \`shortcuts/doc/table_widths.go\`
  - \`extractMarkdownTables\` – GFM pipe-table parser (handles fenced code blocks, escaped \`\\|\`, multiple tables)
  - \`computeWidthRatios\` – column weights from max visual width per column, apportioned to 100 with rounding error absorbed by the widest column
  - \`visualWidth\` – CJK / emoji / full-width forms count 2, narrow forms 1, control chars 0
  - \`applyMarkdownTableColumnWidths\` – walks the doc's blocks, matches tables by index+column count, PATCHes each; non-fatal on per-table failure
  - \`fetchDocumentTableBlocks\` – paginated walk over \`/open-apis/docx/v1/documents/{id}/blocks\` filtering \`block_type == 31\`
- Wired into \`docs +create\` and \`docs +update --mode overwrite\` behind a new \`--auto-table-widths\` bool flag (default \`true\`, opt-out with \`=false\`)
- Tests in \`shortcuts/doc/table_widths_test.go\` cover parsing, ratio calc, visual-width edge cases; all pass

## Why only overwrite for \`+update\`

For \`append\` / \`insert_*\` / \`replace_range\`, only some table blocks in the document are newly created and identifying them from the MCP response is ambiguous (the response does not return the new block_ids). Rather than patch existing tables incorrectly, those modes ignore \`--auto-table-widths\` for now. Follow-up: a before/after table-count diff could cover \`append\`.

## Test Plan

- [x] \`go test ./shortcuts/doc/...\` passes (new tests cover table parsing, ratio calculation, visual-width edge cases)
- [x] \`go run github.com/golangci/golangci-lint/v2/cmd/golangci-lint@v2.1.6 run --new-from-rev=origin/main ./shortcuts/doc/...\`: 0 issues
- [x] \`go mod tidy\` produced no changes
- [x] Manual sanity: before this PR, \`lark-cli docs +create --markdown file.md\` with an uneven table yielded \`column_width:[244,244,244]\`; the new flow PATCHes \`width_ratios\` proportional to content width. (The current implementation writes \`update_grid_column_width_ratio\` via the docx open-apis; the block's \`column_width\` column values are adjusted on subsequent renders.)

## Related

- No existing issues found via \`gh issue list --repo larksuite/cli --search "column width OR table width"\`

## Notes

- Added flag is opt-out-able to keep the door open for anyone depending on equal widths.
- \`applyMarkdownTableColumnWidths\` treats fetch/PATCH failures as non-fatal (the main content is already created; column-width adjustment is a cosmetic follow-up). Errors are logged to \`stderr\` with the block_id.

## Summary by CodeRabbit

* **New Features**
  * Added an --auto-table-widths flag (enabled by default) to document create and update commands; after creation/update, Markdown pipe-table column widths are auto-adjusted based on cell content to improve table formatting.
  * Auto-adjust runs for updates in overwrite mode when applicable; it is skipped with a logged message if the document/table cannot be resolved or counts/matching fail.

* **Tests**
  * Added unit tests for Markdown table parsing, visual-width estimation, and width-ratio calculations.

### CODE REVIEW

File: shortcuts/doc/table_widths.go [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+				remote[i].BlockID, err,
+			)
+			continue
+		}
+		patched++
+	}
+	if patched > 0 {
+		fmt.Fprintf(runtime.IO().ErrOut, "column widths applied to %d/%d tables\n", patched, limit)
+	}
+}
```
[the remark this answers is not in the document]
Comment (ejjcc) [reply shortcuts/doc/table_widths.go:443]: Thanks—option 1 adopted in 78c9276. applyMarkdownTableColumnWidths now aborts with a single warning on stderr when len(remote) != len(tables) instead of proceeding with a truncated pairing. Also added test coverage for the blank-line and fence-terminated flush paths in extractMarkdownTables so the blockquote/malformed-separator footguns you flagged fail loudly instead of silently shifting indexes.

**Comment by flyinghanger** [comment]
Addressed the review in [78c9276](https://github.com/larksuite/cli/pull/557/commits/78c9276a0e0cf3ef868ce261e5d087209bb3e9dc):

**Actionable**
- Strict `len(remote) == len(tables)` precondition in `applyMarkdownTableColumnWidths`; on mismatch we now log a single stderr warning and skip the whole pass instead of truncating to `min(local, remote)` and silently misaligning everything downstream.

**Nitpicks handled**
- Extracted `docxBlockTypeTable = 31` alongside `widthRatioTotal`, call site reads `if int(bt) != docxBlockTypeTable`.
- `visualWidth` now treats Cf (ZWJ, LRM, …), Mn (combining marks), and Me (enclosing marks) as zero-width, so ZWJ emoji sequences + U+FE0F variation selectors no longer inflate cell weights.
- Added blank-line-separated and fence-terminated table cases to `TestExtractMarkdownTables`.
- `docs +update --mode overwrite --auto-table-widths=true` on a wiki URL now emits a user-facing "skipped: --doc is not a docx token/URL" notice instead of silently no-op'ing.

**Deferred**
- End-to-end test for `applyMarkdownTableColumnWidths` with a stubbed `CallAPI`: good idea, but `*common.RuntimeContext.CallAPI` is a concrete method, not an interface, so mocking it cleanly needs either a small interface extraction or a thin indirection. I'd rather do that refactor in a separate PR so this one stays narrowly scoped — happy to follow up if you prefer. In the meantime, the strict precondition + the per-block `ColumnSize` guard + the non-fatal PATCH-error path cover the same failure modes in prod; the missing piece is just the fake-driven unit test.

**Comment by flyinghanger** [comment]
R2 review addressed in [6fab603](https://github.com/larksuite/cli/pull/557/commits/6fab6030f4e5f5a1e31f9c62ef3c4d6a4a0ce82c):

- **Silent per-table skip** — `applyMarkdownTableColumnWidths` now prints a per-table diagnostic on stderr with the block id + local/remote column counts when a mismatch causes a skip, so the `N/M tables` summary has an actionable signal for the gap.
- **Require GFM separator row** — `extractMarkdownTables` is now a proper state machine: a pipe-row line is only a header candidate; the table is committed only after the next line is confirmed as a `|---|---|` separator. Added 3 tests covering the footguns (single pipe-ish prose line; two pipe rows without separator; dangling pipe header followed by a real table).
- **Wider emoji / symbol coverage in `isWideRune`** — added Misc Symbols (U+2600-26FF), Dingbats (U+2700-27BF), Mahjong / Playing Cards / Enclosed Alphanumeric (U+1F000-1F2FF), and Regional Indicator Symbols (U+1F1E6-1F1FF). Added `visualWidth` test cases for ☀️ / ✅ / 🀄 / 🇺🇸. Kept the manual ranges rather than importing `golang.org/x/text/width` — the dep was worth a check but isn't in `go.sum` today, and manual ranges keep this change self-contained without touching `go.mod`.

All new tests pass, `golangci-lint --new-from-rev=origin/main ./shortcuts/doc/...` shows 0 issues.

**Comment by flyinghanger** [comment]
Correction to my last comment — `golang.org/x/text` **is** a direct dependency, not just transitive. Followed up in [2692887](https://github.com/larksuite/cli/pull/557/commits/2692887): `isWideRune` now delegates East Asian Width classification to `width.LookupRune` (CJK / Hangul / full-width forms, all of BMP correctly categorised), keeping manual ranges only for the neutral-but-visually-wide blocks (Misc Symbols, Dingbats, Mahjong / Playing Cards / Enclosed Alphanumeric, emoji pictographs, Regional Indicator Symbols). `go mod tidy` still a no-op.

File: shortcuts/doc/table_widths.go [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+		widest := 0
+		for i := 1; i < cols; i++ {
+			if weights[i] > weights[widest] {
+				widest = i
+			}
+		}
+		ratios[widest] += diff
+		if ratios[widest] < 1 {
+			ratios[widest] = 1
+		}
```
Comment (fangshuyu-768) [inline shortcuts/doc/table_widths.go:294]: Nit — correctness: this `<1` clamp *after* the `diff` adjustment silently breaks the documented sum-to-100 invariant. The widest column's ratio is usually large enough that `diff` can't push it below 1, but the clamp exists specifically for the case where it could — and when it fires, the returned slice no longer sums to 100, which is what the API expects.

Constructing a real failure is hard (you'd need many clamped-up tiny columns driving `allocated` far above 100), but if it ever happens the CLI would quietly write an invalid `width_ratios` array. Two options:

1. Drop the clamp here and, if `ratios[widest]+diff < 1`, return `nil` (skip the patch) — safer than writing a bad ratio.
2. Redistribute the leftover deficit across the other columns instead of piling it all on the widest.

Either way, a short assertion-style test (`sum == 100` on a pathological weight distribution) would lock it in.

File: shortcuts/doc/table_widths.go [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+	case r >= 0x2600 && r <= 0x26FF: // Misc Symbols (☀ ☁ ⚡ ♻ …)
+		return true
+	case r >= 0x2700 && r <= 0x27BF: // Dingbats (✅ ✈ ✂ ✏ …)
+		return true
+	case r >= 0x1F000 && r <= 0x1F2FF: // Mahjong / Playing Cards / Enclosed Alphanumeric (🀄 🃏 🅰 🆗 …)
+		return true
+	case r >= 0x1F300 && r <= 0x1FAFF: // Emoji & pictographs
+		return true
+	case r >= 0x1F1E6 && r <= 0x1F1FF: // Regional Indicator Symbols (flag halves)
+		return true
```
Comment (fangshuyu-768) [inline shortcuts/doc/table_widths.go:356]: Minor: Regional Indicator pairs (flags) and emoji ZWJ sequences overcount here. `🇺🇸` yields 4 (two RIs × 2), and `👨‍👩‍👧‍👦` yields 8 (four base emoji × 2, ZWJ=0), but both render as one grapheme cluster of visual width 2. Since these widths are only used as weights for proportional ratios, it's not catastrophic — but a cell that's mostly flags will claim disproportionately more column width than the eye expects.

Not necessarily a blocker (tables of flags are rare), but either (a) a `TODO` comment pointing at `golang.org/x/text/unicode/norm` + grapheme-cluster segmentation, or (b) a test asserting the current (known-overcounting) behavior so it doesn't silently change, would be worth adding.

File: shortcuts/doc/table_widths.go [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+
+// applyMarkdownTableColumnWidths calculates per-column width ratios for each
+// pipe table in the supplied markdown and PATCHes matching table blocks in the
+// given document with update_grid_column_width_ratio. Failures are logged and
+// treated as non-fatal because the main content has already been created.
+//
+// Only tables whose local column count equals the remote table's column_size
+// are patched; mismatches are skipped. Tables are matched to remote table
+// blocks by document-order index.
+func applyMarkdownTableColumnWidths(runtime *common.RuntimeContext, documentID, markdown string) {
```
Comment (fangshuyu-768) [inline shortcuts/doc/table_widths.go:369]: Coverage gap: `extractMarkdownTables`, `computeWidthRatios`, and `visualWidth` all have solid unit tests, but `applyMarkdownTableColumnWidths` — where the actual risk lives (index-based matching, strict remote-vs-local count check, per-table PATCH-failure handling, `column_size` mismatch skip) — has none.

A table-driven test with a fake `runtime.CallAPI` (or a narrow interface around the blocks fetch + PATCH) would let you pin:

- skip when `len(remote) != len(tables)`,
- skip (and continue) when `column_size` differs for a given pair,
- non-fatal on per-block PATCH error,
- exit status unchanged on any failure.

Given this is the subtlest part of the change and runs on every `+create`/overwrite, I'd strongly prefer at least one integration-style test here.

File: shortcuts/doc/docs_create.go [hunk 5]
```diff
@@ -61,6 +62,11 @@ var DocsCreate = common.Shortcut{
 		augmentDocsCreateResult(runtime, result)
 
 		normalizeDocsUpdateResult(result, runtime.Str("markdown"))
+		if runtime.Bool("auto-table-widths") {
+			if docID := strings.TrimSpace(common.GetString(result, "doc_id")); docID != "" {
+				applyMarkdownTableColumnWidths(runtime, docID, runtime.Str("markdown"))
```
Comment (fangshuyu-768) [inline shortcuts/doc/docs_create.go:67]: Consider validating `doc_id` is actually a docx token before calling into the docx blocks API. For `--wiki-node` / `--wiki-space` creates, if the MCP response ever surfaces a wiki token (or anything non-docx) in `doc_id`, the subsequent `GET /open-apis/docx/v1/documents/{id}/blocks` 404s and we emit an error-shaped log line — which is then non-fatal per design, but reads like a real failure to the user.

Cleaner: pass `doc_url` (or `doc_id`) through `parseDocumentRef`, and only proceed if `ref.Kind == "docx"`. That mirrors what `selectDocsPermissionTarget` already does and produces a clean skip message instead of an API error log in the wiki-create path.

File: shortcuts/doc/docs_update.go [hunk 6]
```diff
@@ -112,6 +114,16 @@ var DocsUpdate = common.Shortcut{
...
 
 		normalizeDocsUpdateResult(result, runtime.Str("markdown"))
+		if runtime.Str("mode") == "overwrite" && runtime.Bool("auto-table-widths") {
+			if docID, ok := docxTokenForUpdate(runtime.Str("doc")); ok {
+				applyMarkdownTableColumnWidths(runtime, docID, runtime.Str("markdown"))
+			} else {
+				fmt.Fprintln(
+					runtime.IO().ErrOut,
+					"auto-table-widths skipped: --doc is not a docx token/URL (wiki-backed docs require an extra resolve step that is not implemented yet)",
+				)
```
Comment (fangshuyu-768) [inline shortcuts/doc/docs_update.go:124]: The skip message is honest, but `docs +update` already accepts wiki URLs end-to-end in the main path (the MCP `update-doc` handles them), so any user who habitually passes `/wiki/...` URLs will always hit this skip on overwrite — silently getting the old equal-width behavior while the create path auto-adjusts. That's an easy-to-miss inconsistency.

There's already a wiki→obj resolution pattern in `shortcuts/wiki` (`/open-apis/wiki/v2/spaces/get_node`) returning the `obj_token` / `obj_type`. One extra call on the overwrite path would let you handle wiki URLs uniformly instead of punting. If you'd rather not expand scope in this PR, please at least link a follow-up issue — the PR description mentions this as future work but there's no tracking link.

File: shortcuts/doc/docs_create.go [hunk 7]
```diff
@@ -23,6 +23,7 @@ var DocsCreate = common.Shortcut{
 		{Name: "folder-token", Desc: "parent folder token"},
 		{Name: "wiki-node", Desc: "wiki node token"},
 		{Name: "wiki-space", Desc: "wiki space ID (use my_library for personal library)"},
+		{Name: "auto-table-widths", Type: "bool", Default: "true", Desc: "after create, adjust Markdown pipe-table column widths based on cell content (use --auto-table-widths=false to keep server default equal widths)"},
```
Comment (fangshuyu-768) [inline shortcuts/doc/docs_create.go:26]: The default is `true`, which changes the rendered shape of every newly-created/overwritten doc with tables. Users with scripted workflows (golden fixtures, screenshot regressions, reviewer tooling that diffs rendered docs) will see output change without opting in.

The opt-out knob is clearly documented in the flag description — good — but I'd also call this behavior change out explicitly in the CHANGELOG / release notes so it doesn't surprise anyone who doesn't read flag help. Not blocking.

**Review by fangshuyu-768 — COMMENTED** [review]
Nice feature — the core parser/ratio logic is well-scoped and the tests cover the tricky GFM edge cases. A few things worth a second look before merge: one correctness nit in `computeWidthRatios` (sum-to-100 invariant can break under the `<1` clamp), the wiki-URL gap on `docs +update`, over-counting of ZWJ/flag emoji in `visualWidth`, and no tests around the orchestration path (where index-matching and per-table failures actually live). Nothing blocking if the team is fine with those caveats; inline comments below.

File: shortcuts/doc/table_widths_test.go [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+		}
+	})
+
+	t.Run("table inside fenced code is skipped", func(t *testing.T) {
+		md := "```md\n| A | B |\n|---|---|\n| 1 | 2 |\n```\n"
+		got := extractMarkdownTables(md)
+		if len(got) != 0 {
+			t.Fatalf("expected 0 tables inside fence, got %d", len(got))
+		}
+	})
```
[the remark this answers is not in the document]
Comment (ejjcc) [reply shortcuts/doc/table_widths_test.go:81]: Addressed in ae21ab0: added `longer fence is not closed by a shorter inner fence` sub-test and extracted fence scanning into a `detectFenceMarker` helper that tracks the opening char + run length, so only a matching-or-longer marker of the same char closes the fence.

File: shortcuts/doc/table_widths_test.go [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+				if r < 1 {
+					t.Errorf("sample %d: ratio < 1 in %v", i, got)
+				}
+				sum += r
+			}
+			if sum != 100 {
+				t.Errorf("sample %d: sum != 100 (got %d) from %v", i, sum, got)
+			}
+		}
+	})
```
[the remark this answers is not in the document]
Comment (ejjcc) [reply shortcuts/doc/table_widths_test.go:242]: Addressed in ae21ab0: `computeWidthRatios` now returns nil when `cols > widthRatioTotal` (commit 53771a7 already added the guard; this pass adds explicit regression tests: `nil when column count exceeds widthRatioTotal` and `exactly widthRatioTotal columns sum to 100`, plus a pathological many-skinny-columns sample in `sum always equals 100 across varied shapes` to pin the apportionment behaviour).

File: shortcuts/doc/table_widths.go [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+			// current line as a new candidate (it might still be a header
+			// for a table that follows).
+			header = splitPipeRow(trimmed)
+		case current != nil:
+			// Inside a confirmed table.
+			if isPipeSeparatorRow(trimmed) {
+				// Separator rows after the confirmed one are unusual; drop.
+				continue
+			}
+			current = append(current, splitPipeRow(trimmed))
```
[the remark this answers is not in the document]
Comment (ejjcc) [reply shortcuts/doc/table_widths.go]: Addressed in ae21ab0: the parser now tracks the confirmed column count, rejects separator rows whose cell count disagrees with the pending header (drops the header entirely), and normalises body rows (truncate/pad) to the confirmed width so `computeWidthRatios` receives a ratio list that matches `column_size`. New sub-tests: `separator with different cell count rejects the header` and `body rows are normalised to the confirmed column count`.

File: shortcuts/doc/table_widths.go (hunk 2 above)
Comment (ejjcc) [reply shortcuts/doc/table_widths.go:294]: Went with option 1 in ae21ab0: if `ratios[widest]+diff < 1` after the apportionment adjustment, `computeWidthRatios` now returns nil rather than silently writing an invalid `width_ratios` array. Added a pathological sample (one wide column + 49 skinny ones) to the `sum always equals 100` subtest so future regressions trip the assertion. Combined with the new `cols > widthRatioTotal` short-circuit, both paths out of the invariant are locked down.

File: shortcuts/doc/table_widths.go (hunk 3 above)
Comment (ejjcc) [reply shortcuts/doc/table_widths.go:356]: Added explicit documentation of the known overcount on `visualWidth` in ae21ab0 (Regional-Indicator pairs count 4 instead of 2, four-person ZWJ family counts 8 instead of 2) pointing at grapheme-cluster segmentation (rivo/uniseg) as the follow-up. The existing `visualWidth` test cases now pin the current behaviour (`🇺🇸` → 4) so any change to this policy trips a failing test. Kept it in-tree rather than pulling a new dep for a proportional-weighting edge case.

File: shortcuts/doc/table_widths.go (hunk 4 above)
Comment (ejjcc) [reply shortcuts/doc/table_widths.go:369]: Addressed in ae21ab0: added `TestApplyTableColumnWidths` with a `fakeAPI` stubbing `runtime.CallAPI`. Covers happy path, whole-doc count mismatch, per-table `column_size` mismatch, non-fatal per-block PATCH error, and blocks-fetch error. Also added `TestDocxTokenForAutoWidths` / `TestResolveDocxTokenForCreateResult` for the token resolvers. The orchestration code was refactored to accept an internal `apiCaller` function type that `*common.RuntimeContext.CallAPI` already satisfies, so the public entry points didn't change.

File: shortcuts/doc/docs_create.go (hunk 5 above)
Comment (ejjcc) [reply shortcuts/doc/docs_create.go:67]: Addressed in ae21ab0: `docs +create` now routes the docx token via `resolveDocxTokenForCreateResult`, which prefers the MCP result's `doc_url` + `parseDocumentRef` (same pattern as `selectDocsPermissionTarget`). Wiki URLs take one `/open-apis/wiki/v2/spaces/get_node` hop and fall back to `doc_id` only when no `doc_url` is returned. No more spurious 404 on wiki-backed creates.

File: shortcuts/doc/docs_update.go (hunk 6 above)
Comment (ejjcc) [reply shortcuts/doc/docs_update.go:124]: Addressed in ae21ab0: `docs +update --mode overwrite` now resolves wiki URLs to their backing docx via the same `/open-apis/wiki/v2/spaces/get_node` call, so `--auto-table-widths` behaves uniformly across create and update, docx and wiki. The skip path only fires for genuinely non-docx inputs (e.g. sheet-backed wiki nodes), and its message was trimmed to 'could not resolve --doc to a docx token'.

File: shortcuts/doc/docs_create.go (hunk 7 above)
Comment (ejjcc) [reply shortcuts/doc/docs_create.go:26]: Fair point — keeping the default as `true` because the old equal-width rendering is often visibly worse and the opt-out is documented in the flag description, but the behaviour change will be called out in the release notes when this ships. I didn't edit CHANGELOG.md directly because the repo's convention is to accumulate release notes in the release-cut PR (see the v1.0.14 section header); happy to add an explicit CHANGELOG entry here if maintainers prefer.

**Comment by flyinghanger** [comment]
Third review pass (coderabbit R3/R4 + @fangshuyu-768) addressed in [ae21ab0](https://github.com/larksuite/cli/pull/557/commits/ae21ab0).

Also picked up commit 53771a7 from the earlier scheduled pass via rebase — its fence-marker / >100-col handling is retained but extended with row-normalisation, separator-cell-count validation, and additional tests.

### Correctness
- `computeWidthRatios` returns nil for `cols > widthRatioTotal` (locked by two new tests); also returns nil if the final rounding adjustment would push the widest column's ratio below 1, so the ≥1-per-column floor and the sum-to-100 invariant can never disagree.
- `extractMarkdownTables` tracks the opening fence char + length, rejects separator rows whose cell count disagrees with the pending header, and normalises body rows to the confirmed column count.

### Orchestration & wiki support
- `docs +create` now routes the docx token via the MCP result's `doc_url` + `parseDocumentRef` (same pattern as `selectDocsPermissionTarget`), taking one `/open-apis/wiki/v2/spaces/get_node` hop when needed. No more spurious 404 on wiki-backed creates.
- `docs +update --mode overwrite` resolves wiki URLs the same way, so `--auto-table-widths` behaves uniformly between create and update across docx / wiki inputs.

### Testability
- Introduced an internal `apiCaller` function type that `*common.RuntimeContext.CallAPI` satisfies; `applyMarkdownTableColumnWidths` / `fetchTableBlocks` / `docxTokenForUpdate` / `resolveDocxTokenForCreateResult` now take it as a seam. Public entry points unchanged.
- New `TestApplyTableColumnWidths` with a `fakeAPI` covers: happy path, whole-doc count mismatch, per-table column-size skip with diagnostic, non-fatal per-block PATCH error, blocks-fetch error.
- New `TestDocxTokenForAutoWidths` and `TestResolveDocxTokenForCreateResult` cover docx short-circuit, wiki resolution, non-docx wiki node skip, resolve error.

### Docs
- Added a TODO on `visualWidth` explaining the known Regional-Indicator / ZWJ overcount; existing visualWidth cases pin the current behaviour so any future policy change trips a failing test.

All new tests pass; `golangci-lint --new-from-rev=origin/main` shows 0 issues; `go mod tidy` is a no-op.

**Comment by SunPeiYang996** [comment]
Thanks for the work here. I’m going to close this PR rather than merge it.

This area was already handled in the earlier table column-width work from #59. The final direction there was that the new table column-width algorithm had already shipped to `main` through a separate change and is available in the latest `lark-cli`. This PR would duplicate the same capability through a different implementation path.

The algorithm we want to keep from #59 is roughly as follows.

## Core Idea

The column-width problem is modeled as constrained resource allocation:

- **Resource**: usable page width, `pageWidth × fillRatio`
- **Subject**: each column is estimated with three width tiers: `min` / `pref` / `max`
- **Policy**: satisfy every column’s minimum width first, then fit natural preferred widths, and only expand into the remaining budget when the content actually warrants it

Columns are also differentiated by content type: text columns are more likely to expand, while numeric / boolean / enum-like columns stay compact.

## Three-Tier Width Estimation

For every column, the algorithm scans all cells and computes:

- **min**: longest unbreakable token + cell padding, so words are not hard-broken
- **pref**: a bucketed quantile over per-row content length, weighted by the coefficient of variation; the header is included with a lower weight so a long header alone cannot dominate a column of short cells
- **max**: a cap based on page width, further clamped by column type

A glyph-width safety factor is also applied to compensate for systematic under-estimation in the internal width oracle.

## Five-Pass Pipeline

```text
function LayoutColumns(table, pageWidth):
    if table.hasExplicitColWidth():
        return                              # respect existing widths

    budget := pageWidth * fillRatio
    cols   := []
    for c in table.columns:
        cols[c] := estimate(c)              # min/pref/max + column type

    # Pass 1: shrink if sum(pref) > budget
    while sum(pref) > budget:
        i := argmax(pref)
        pref[i] := nextLargerNeighbor(pref)

    # Pass 2: expand if sum(pref) < budget
    if mode == ExpandToBudget:
        distribute(budget - sum(pref)) by columnTypeWeight
    else:
        pref[i] *= softExpandFactor         # gentle expansion, do not force-fill

    # Pass 3: enforce min, iterate to a fixed point
    repeat N times:
        for i where pref[i] < min[i]:
            pref[i] := min[i]
        redistribute remaining budget

    # Pass 4: smooth neighbours
    for adjacent pair (i, j) where ratio(i, j) > threshold:
        borrow from wider to narrower

    # Pass 5: round + clamp
    for each col:
        w[i] := ceil(pref[i])
        w[i] := clamp(w[i], hardMin, max[i])

    return w
```

## Column-Type Differentiation

| Type | Characteristics | Strategy |
|---|---|---|
| Text | Large length variance, breakable | High weight, biased to expand |
| Code | Unbreakable, monospaced | Medium weight, slightly wider max |
| Numeric | Short, uniform | Low weight, conservative expand |
| Boolean/Enum | Very short | Excluded from expansion |

## Edge Cases

- **Two-column table with a dominant column**: when one column’s preferred width is much larger than the other, its `max` is relaxed so a long description is not hard-capped.
- **Very wide tables**: when even `min × N` exceeds the budget, the total-width constraint is dropped and every column gets its `min`; horizontal scrolling is left to the front-end.
- **Streaming / incomplete input**: column-width computation is skipped until the table structure stabilizes.

This design is more robust than the implementation path in this PR, which parses local Markdown pipe tables after create/update, matches remote table blocks by order and column count, and then applies an additional PATCH for width ratios. That post-processing path is vulnerable to read-after-write consistency issues, table matching drift, and extra PATCH latency.

Since the intended algorithm is already available on `main`, this PR would introduce a second, weaker implementation of the same behavior. Closing it to avoid maintaining two table-width behaviors.
